client/client.go

```diff
@@ -733,9 +737,9 @@ func (c *Client) reloadTLSConnections(newConfig *nconfig.TLSConfig) error {
 
 	// Keep the client configuration up to date as we use configuration values to
 	// decide on what type of connections to accept
-	c.configLock.Lock()
-	c.config.TLSConfig = newConfig
-	c.configLock.Unlock()
+	c.UpdateConfig(func(c *config.Config) {
+		c.TLSConfig = newConfig
+	})
```

Comment:
Here's another pattern you should see a lot (and I call out in the configLock comment): using the Client.UpdateConfig helper when mutating the config. It handles the locking and copying aspects of safely manipulating the config at runtime.